packages/Search/src/details/DTK_DetailsTmp.hpp

```diff
@@ -0,0 +1,690 @@
+        KOKKOS_LAMBDA( int q ) {
+            for ( int i = offset( q ); i < offset( q + 1 ); ++i )
+            {
+                export_ranks( i ) = ranks( q );
+            }
+        } );
+    Kokkos::fence();
+
+    auto distributor = std::make_shared<Tpetra::Distributor>( comm );
+    int const n_imports = distributor->createFromSends(
+        Teuchos::ArrayView<int>( export_ranks.data(), n_exports ) );
+
+    // export_ranks already has adequate size since it was used as a buffer to
+    // make the new communication plan.
+    Kokkos::parallel_for(
+        REGION_NAME( "communicate_results_back_fill_rank" ),
+        Kokkos::RangePolicy<ExecutionSpace>( 0, n_exports ),
+        KOKKOS_LAMBDA( int i ) { export_ranks( i ) = comm_rank; } );
+    Kokkos::fence();
+
+    Kokkos::View<int *, DeviceType> export_ids( "export_ids", n_exports );
+    Kokkos::parallel_for(
+        REGION_NAME( "communicate_results_back_fill_ids" ),
+        Kokkos::RangePolicy<ExecutionSpace>( 0, n_fwd_queries ),
+        KOKKOS_LAMBDA( int q ) {
+            for ( int i = offset( q ); i < offset( q + 1 ); ++i )
+            {
+                export_ids( i ) = ids( q );
+            }
+        } );
+    Kokkos::fence();
+    Kokkos::View<int *, DeviceType> export_indices = indices;
+
+    Kokkos::View<int *, DeviceType> import_indices( "import_indices",
+                                                    n_imports );
+    Kokkos::View<int *, DeviceType> import_ranks( "import_ranks", n_imports );
+    Kokkos::View<int *, DeviceType> import_ids( "import_ids", n_imports );
+    send_across_network( distributor, export_indices, import_indices );
+    send_across_network( distributor, export_ranks, import_ranks );
+    send_across_network( distributor, export_ids, import_ids );
+
+    ids = import_ids;
+    ranks = import_ranks;
+    indices = import_indices;
+}
+
+template <typename DeviceType>
+void Tmp<DeviceType>::communicate_results_back(
+    Teuchos::RCP<Teuchos::Comm<int> const> comm,
+    Kokkos::View<int *, DeviceType> &indices,
+    Kokkos::View<int *, DeviceType> offset,
+    Kokkos::View<int *, DeviceType> &ranks,
+    Kokkos::View<int *, DeviceType> &ids,
+    Kokkos::View<double *, DeviceType> &distances )
+{
+    int const comm_rank = comm->getRank();
+
+    int const n_fwd_queries = offset.extent_int( 0 ) - 1;
+    int const n_exports = offset( n_fwd_queries );
+    Kokkos::resize( ranks, n_exports );
+    Kokkos::View<int *, DeviceType> export_ranks( "", n_exports );
+    Kokkos::parallel_for(
+        REGION_NAME( "communicate_back_setup_plan" ),
+        Kokkos::RangePolicy<ExecutionSpace>( 0, n_fwd_queries ),
+        KOKKOS_LAMBDA( int q ) {
+            for ( int i = offset( q ); i < offset( q + 1 ); ++i )
+            {
+                export_ranks( i ) = ranks( q );
+            }
+        } );
+    Kokkos::fence();
+
+    auto distributor = std::make_shared<Tpetra::Distributor>( comm );
```

Comment:
No there is no assignment operator with `Tpetra::Distributor` and they are not needed any more after.